lib/daru/view/adapters/highcharts/display.rb

```diff
@@ -61,6 +64,52 @@ def to_html_iruby(placeholder=random_canvas_id)
       high_chart_iruby(extract_chart_class, placeholder, self)
     end
 
+    # Loads the dependent mapdata and dependent modules of the chart
+    #
+    # @param [String] to determine whether to load modules in IRuby or web
+    #   frameworks
+    # @return [void, String] loads the initial script of the modules for IRuby
+    #   notebook and returns initial script of the modules for web frameworks
+    def load_dependencies(type)
+      dep_js = extract_dependencies
+      if type == 'iruby'
+        LazyHighCharts.init_iruby(dep_js) unless dep_js.nil?
+      elsif type == 'web_frameworks'
+        dep_js.nil? ? '' : LazyHighCharts.init_script(dep_js)
+      end
+    end
+
+    # Extracts the required dependencies for the chart. User does not need
+    #   to provide any mapdata requirement explicity in the `options`.
+    #   MapData will be extracted using `options[:chart][:map]` already
+    #   provided by the user. In `modules` user needs to provide the required
+    #   modules (like tilemap in highcharts) in the form of Array. Once the
+    #   dependency is loaded on a page, there is no need to provide it again in
+    #   the `modules` option.
+    #
+    # @return [Array] the required dependencies (mapdata or modules)
+    #   to load the chart
+    def extract_dependencies
+      dep_js = []
+      # Mapdata dependencies
+      get_map_data_dependencies(dep_js)
+      # Dependencies provided in modules option (of highcharts mainly
+      #   like tilemap) by the user
+      dep_js |= options.delete(:modules).map! { |js| "#{js}.js" } unless
+      options[:modules].nil?
+      dep_js
+    end
+
+    # @param dep_js [Array] JS dependencies required for drawing a map(mapdata)
+    # @return [void] Appends the map data in dep_js
+    def get_map_data_dependencies(dep_js)
```

Comment:
According to HighMap option `map`, we are loading the respective js files (so no need to load not used files). 

But need to find a way to capture the loaded js files, so that next time we run `to_html` or generate the script it won't load the file, that is already loaded into the **same  web-page**.